mitxpro/models.py

```diff
@@ -52,3 +55,69 @@ class TimestampedModel(Model):
 
     class Meta:
         abstract = True
+
+
+class AuditModel(TimestampedModel):
+    """An abstract base class for audit models"""
+
+    acting_user = ForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete=PROTECT)
+    data_before = JSONField(blank=True, null=True)
+    data_after = JSONField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def get_related_field_name(cls):
+        """
+        Returns:
+            str: A field name which links the Auditable model to this model
+        """
+        raise NotImplementedError
```

Comment:
Take or leave: Would it be better to require that classes return a field here instead of a field name? e.g.: `return self.order` instead of `return "order"`. The return type here would then be `django.db.models.fields.Field`, and those `Field` objects have a `name` property that could give you the name as a string. One advantage: if we needed to change the name of a field referred to by `get_related_field_name`, an IDE could do that without an issue if the reference was `self.<field_name>`

Reply:
To aid in IDE inspection we could change the implementation to return the field name based on the field object reference. We wouldn't need to change this function for that.